docs/patterns/CVE-2020-35892-3-MIR.rpl

```diff
@@ -0,0 +1,51 @@
+pattern CVE-2020-35892-3
+
+patt {
+    use core::ops::range::Range<usize>;
+
+    // how to express constraints on meta variables?
+    // 1. use a where clause (how to define a where clause?)
+    // 2. use a predicate (how to define a predicate?)
+    p[
+        $T: ty,
+        $SlabT: ty
+    ] where $SlabT: Adt 
```

Comment:
Here is a question about RPL design:
How to express constraints on meta variables?
1. use a where clause (how to define a where clause?)
2. use a predicate (how to define a predicate?)
@frank-king @TheVeryDarkness

Reply:
If the meta variable has some constraints, I think we should define a new pattern instead.